### ISSUE TITLE
Is it safe to use wip-edison-3.19.5 branch?

### ISSUE THREAD

**Issue by sandeepmvd** [intel/edison-linux#22]
Hi,
Is it ok and is it safe to use wip-edison-3.19.5 branch for edison? When can we expect the official release?

**Comment by andy-shev** [comment]
I'm not sure WIP is a correct abbreviation there. Last commit dated end of May last year. You may instead try my unofficial **eds** branch (see my GH page, **linux** repository).

**Comment by sandeepmvd** [comment]
Thanks @andy-shev 

The kernel on eds branch is 4.10. Are the edison patches ported to it?
If yes, it would be awesome! :+1: 

I will try and post the results.. Thanks...

**Comment by andy-shev** [comment]
@sandeep-mvd
Not everything is working there. There are a lot of areas out of my scope. But some necessary minimum works fine. I would suggest to follow the discussion on https://communities.intel.com/thread/75472.

**Comment by agherzan** [comment]
I see now a branch that doesn't have the wip prefix. Is that supposed to mean that it is stable to be used?

**Comment by andy-shev** [comment]
@agherzan May I ask what is your use case for that old kernel?

**Comment by agherzan** [comment]
I don't have a usecase of an old kernel. I'm looking for a stable one that is officially supported.

**Comment by andy-shev** [comment]
Fair enough. As far as I know the latest that officially supported is based on v3.10.98.

**Comment by agherzan** [comment]
Is there any documentation on this? From the branching names 3.19 looks stable.

**Comment by andy-shev** [comment]
I dunno about documentation, I looked at the code, basically latest BSP for Edison. There is still v3.10.98 based kernel which makes my conclusion.

**Comment by agherzan** [comment]
So you are saying that edison-3.19 is not a stable branch, right?

**Comment by agherzan** [comment]
I'm really confused about 3.10.98. I try to apply aufs patches on top. but things fail because it seems this branch includes patches which were merged in later versions of kernel:
https://github.com/torvalds/linux/commit/946e51f2bf37f1656916eb75bd0742ba33983c28
https://github.com/torvalds/linux/commit/eee5cc2702929fd41cce28058dc6d6717f723f87

The nice thing about edison-3.19 is that is already including aufs so it works out of the box but I don't know the dev status of this branch.

**Comment by htot** [comment]
edison-3.19.5 has last commit 4-jun-2015. wip-edison-3.19.5 has last commit 10-may-2016. Comparing them shows wip has about 64 additional commits in a one year period. You could say that both are pretty stable. Or dead.

**Comment by andy-shev** [comment]
@htot I like your ironic mood! :+1:

**Comment by agherzan** [comment]
@htot Sadly it sounds right :+1: Are there any maintainers of this repo who can confirm/inform?

**Comment by htot** [comment]
:-) I didn't want to hurt anybodies feelings. The difference between stable, finished code (no commits over time) and dead abandoned code (no commits over time) is only if it works for you or not.
I just read this article [https://www.linux.com/news/event/ELCE/2017/long-term-embedded-linux-maintenance-made-easier](url) and concluded that it would probably be best that the work from this repository be integrated into upstream as soon as possible. I am really hoping that behind the scenes Intel developers are supporting Andy's work to bring Edison support into main line. That would probably be more productive than continue working on out-of-tree 3.10 or 3.19 (note that there is no preempt_rt for 3.19 and never will be, so for industrial use 3.19 will in most cases not be an appropriate choice).

**Comment by agherzan** [comment]
Currently I'm only interested in aufs support so 3.19 fixes my incompatibility but I totally agree with what you say.

**Comment by htot** [comment]
I don't know aufs. But when I applied preempt-rt to the 3.10.14 kernel I needed to disable aufs in the kernel config. So may be it has been backported into 3.10.14?

**Comment by agherzan** [comment]
Sounds like a mess...

**Comment by sandeepmvd** [comment]
Well with no option left, I finally had to ditch edison and move to beaglebone.. I can close this issue if no one has anything more to discuss.

**Comment by agherzan** [comment]
I think we can leave it open. Nothing was resolved.

**Comment by andy-shev** [comment]
@agherzan, I don't think it will be for any of v3.xx kernel. So, agree on @binary-nerd, issue can be closed.

**Comment by agherzan** [comment]
OK. We, at resin, are planning to drop support for this board anyway. So I guess I'm fine. (sad story)

**Comment by andy-shev** [comment]
@agherzan 
I dunno what *resin* means, though v4.12-rc1 of the kernel supports most of the stuff on Edison, which means any Linux distro can be run on it. Thus, the obvious question why are you so linked to Yocto based disto for this board?

**Comment by agherzan** [comment]
@andy-shev resin means https://resinos.io/ . And we use yocto to compile the same distro for multiple board types.

**Comment by andy-shev** [comment]
@agherzan 
I see. So, what prevents you to switch to vanilla kernel instead in meta-edison or whatever layer you have (I suppose you have layer per platform anyway) and build same distribution for all boards?

**Comment by andy-shev** [comment]
Since we have established https://github.com/edison-fw/meta-intel-edison for more than year, I think this question doesn't make sense anymore.
